plugins/contact.py

```diff
@@ -0,0 +1,101 @@
+""" A bot to make the wordkeeper's job a little easier during games of Contact.
+
+wordkeeper> !challenge @player1 @player2
+slaskbot> @player1 @player2: type !ready when here
+player1> !ready
+player2> !ready
+slaskbot> (@wordkeeper) Starting countdown...
+slaskbot> 3
+slaskbot> 2
+slaskbot> 1
+slaskbot> go!
+
+"""
+
+import logging
+import re
+import time
+
+import config
+from models.contact import Challenge
+
+
+# Number of seconds we wait for someone to be ready
+TIMEOUT = 120
+
+
+def on_message(msg, server):
+    channel = server["channel"]
+    if channel not in config.CONTACT_CHANNELS:
+        return
+
+    with Challenge.lock(channel) as challenge:
+        if re.search(r"!contactreset", msg.get("text", "")):
+            challenge.reset(channel)
+            return "contactbot reset"
+
+        state = challenge.state
+        if state == Challenge.COMPLETED:
```

Comment:
Mk. Now that I've removed `!ready`, there's only one command, `!challenge`, and I'll make it so that that always puts the bot into challenge mode. That should stop us from getting stuck, too.